spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/MessageListenerAdapter.java

```diff
@@ -344,6 +344,25 @@ protected String getListenerMethodName(Message originalMessage, Object extracted
 		return new Object[] {extractedMessage};
 	}
 
+	/**
+	 * Build an array of arguments to be passed into the target listener method. Allows for multiple method arguments to
+	 * be built from message object with channel, More detail about extractedMessage in the method
+	 * #buildListenerArguments(java.lang.Object)
+	 * <p>
+	 * This can be overridden to treat special message content such as arrays differently, and add argument in case of
+	 * receiving Channel and original Message object to invoke basicAck method in the listener by manual acknowledge
+	 * mode
+	 *
+	 * @param extractedMessage the content of the message
+	 * @param channel the Rabbit channel to operate on
+	 * @param message the incoming Rabbit message
+	 * @return the array of arguments to be passed into the listener method (each element of the array corresponding to
+	 * a distinct method argument)
+	 */
+	protected Object[] buildListenerArguments(Object extractedMessage, Channel channel, Message message) {
+		return buildListenerArguments(extractedMessage);
```

Comment:
Do we deprecate this old method or not?

For end-users it is still going to be a confusing which one to implement.

Reply:
Yes, we should deprecate it.